db/crud.go

```diff
@@ -489,8 +489,22 @@ func (db *DatabaseContext) getRevision(doc *document, revid string) (Body, error
 	body[BodyId] = doc.ID
 	body[BodyRev] = revid
 
-	if doc.Attachments != nil {
-		body[BodyAttachments] = doc.Attachments
+	if doc.CurrentRev == revid && doc.Attachments != nil {
+		attachments := make(map[string]interface{})
+		revGeneration, _ := base.ToInt64(genOfRevID(revid))
+		for name, attachment := range doc.Attachments {
+			attachmentMeta, ok := attachment.(map[string]interface{})
+			if ok {
+				attachmentRevpos, ok := base.ToInt64(attachmentMeta["revpos"])
+				if ok && attachmentRevpos <= revGeneration {
+					attachments[name] = attachment
+				}
+			}
+		}
+
+		if len(attachments) > 0 {
+			body[BodyAttachments] = attachments
+		}
```

Comment:
This work shouldn't be needed.  If doc.CurrentRev==revid, then doc.Attachments should already contain the attachments.  Iterating over the attachments and comparing revpos to revgeneration shouldn't be necessary, as all we're trying to do is move the metadata back as-is from doc.Attachments to body["_attachments"]. The bug was that we were copying this over even if we weren't getting the current revision, and so when requesting a non-winning revision, we were injecting the attachments from that revision.

Reply:
Given that, I would expect this only needs to be:
```
if doc.CurrentRev == revid && doc.Attachments != nil {
   body[BodyAttachments] = doc.Attachments
}